lib/msf/base/sessions/command_shell.rb

```diff
@@ -65,11 +67,126 @@ def desc
     "Command shell"
   end
 
+
+  #
+  # List of supported commands.
+  #
+  def commands
+    c = {
+        'help'            => 'Help menu',
+        'background'   => 'Backgrounds the current shell session',
+        'sessions'     => 'Quickly switch to another session',
+    }
+  end
+
+  #
+  # Parse a line into an array of arguments.
+  #
+  def parse_line(line)
+    line.split(' ')
+  end
+
   #
   # Explicitly runs a command.
   #
   def run_cmd(cmd)
-    shell_command(cmd)
+    if cmd == nil
+      return
+    end
+    arguments = parse_line(cmd)
+    method    = arguments.shift
+
+    # Built-in command
+    if commands.key?(method)
+      return run_command(method, arguments)
+    end
+
+    # User input is not a built-in command, write to socket directly
+    shell_write(cmd)
+  end
+
+  def cmd_help(*args)
+    columns = ['Command', 'Description']
+    tbl = Rex::Text::Table.new(
+      'Header'  => 'Meta shell commands',
+      'Prefix'  => "\n",
+      'Postfix' => "\n",
+      'Indent'  => 4,
+      'Columns' => columns,
+      'SortIndex' => -1
+    )
+    commands.each { |key, value|
```

Comment:
Use `do` for multiline.

Reply:
I was a little confused by this, so I chatted with @wvu to understand better.  He's looking for something like this:

```ruby
commands.each do |key, value|
  tbl << [key, value]
end
```

This would save a line, but also make it clear that there's a loop here.  It's also more consistent with the [Ruby style guide](https://github.com/rubocop-hq/ruby-style-guide#single-line-blocks).

Reply:
For added clarity, maybe change `key` and `value` to `command` and `description`. This could also be done with `map`, but it gets confusing quickly.

Reply:
You can also use `add_row` instead of `<<` for max clarity. :P

Reply:
Cool, thank you so much for code review! helped me a lot.
> ~/.rvm/gems/ruby-2.5.1/gems/rex-text-0.2.20/lib/rex/text/table.rb
```
  #
  # Adds a row using the supplied fields.
  #
  def <<(fields)
    add_row(fields)
  end
```
It seems that `<<` is a syntax sugar of `add_row` or something...
should I change `<<` to `add_row`?